src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/BuildWarningsUtilities.cs

```diff
@@ -0,0 +1,632 @@
+
+					foreach (var expectedWarningsMessage in expectedWarningsPerCode.Messages)
+					{
+						Assert.True(actualWarningsPerCode!.Messages.Remove(expectedWarningsMessage),
+							$"Expected warning message '{expectedWarningsMessage}' was not found for the expected warnings file path '{expectedWarningsPerFile.File}' and warning code '{expectedWarningsPerCode.Code}'");
+					}
+
+					Assert.AreEqual(0, actualWarningsPerCode!.Messages.Count,
+						$"Unexpected warning messages detected for the expected warnings file path '{expectedWarningsPerFile.File}' and warning code '{expectedWarningsPerCode.Code}'! Unexpected warning messages are: {string.Join("\n\t\t", actualWarningsPerCode.Messages)}");
+
+					actualWarningsPerFile.WarningsPerCode.Remove(actualWarningsPerCode);
+				}
+
+				Assert.AreEqual(0, actualWarningsPerFile!.WarningsPerCode.Count,
+					$"Unexpected warning codes detected for the expected warnings file path '{expectedWarningsPerFile.File}'! Unexpected warning codes are: {string.Join("\n\t\t", actualWarningsPerFile.WarningsPerCode.Select(c => c.Code).ToList())}");
+
+				actualWarnings.Remove(actualWarningsPerFile!);
+			}
+
+			Assert.AreEqual(0, actualWarnings.Count,
+				$"Unexpected warning files detected! Unexpected warning file paths are: {string.Join("\n\t\t", actualWarnings.Select(f => f.File).ToList())}");
+		}
+
+		#region Expected warning messages
+
+		// IMPORTANT: Always store expected File information as a relative path to the repo ROOT
+		private static readonly List<WarningsPerFile> expectedNativeAOTWarnings = new()
+		{
+			new WarningsPerFile
+			{
+				File = "ILC",
+				WarningsPerCode = new List<WarningsPerCode>
+				{
+					new WarningsPerCode
+					{
+						Code = "IL3050",
+						Messages = new List<string>
+						{
+							"<Module>..cctor(): Using member 'System.Enum.GetValues(Type)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. It might not be possible to create an array of the enum type at runtime. Use the GetValues<TEnum> overload or the GetValuesAsUnderlyingType method instead.",
+						}
+					},
+				}
+			},
+			new WarningsPerFile
+			{
+				File = "src/Controls/src/Core/ResourceDictionary.cs",
+				WarningsPerCode = new List<WarningsPerCode>
+				{
+					new WarningsPerCode
+					{
+						Code = "IL2067",
+						Messages = new List<string>
+						{
+							"Microsoft.Maui.Controls.ResourceDictionary.<>c.<SetAndLoadSource>b__8_0(Type): 'type' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicParameterlessConstructor' in call to 'System.Activator.CreateInstance(Type)'. The parameter 'key' of method 'Microsoft.Maui.Controls.ResourceDictionary.<>c.<SetAndLoadSource>b__8_0(Type)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.",
```

Comment:
> ResourceDictionary.<>c.<SetAndLoadSource>b__8_0(Type): 'type'

If `b__8_0` changed to `b__9_0`, for example, what would the test failure look like? Would it be easy for a random dev to understand?

Reply:
If `b__9_0 ` starts appearing in the warning message the test would fail with:
```
Expected warning message 'Microsoft.Maui.Controls.ResourceDictionary.<>c.<SetAndLoadSource>b__8_0(Type): 'type' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicParameterlessConstructor' in call to 'System.Activator.CreateInstance(Type)'. The parameter 'key' of method 'Microsoft.Maui.Controls.ResourceDictionary.<>c.<SetAndLoadSource>b__8_0(Type)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.' was not found for the expected warnings file path 'src/Controls/src/Core/ResourceDictionary.cs' and warning code 'IL2067'
  Expected: True
  But was:  False
```
Not sure how intuitive it is, but I tried to include as much information as possible.
Nevertheless, I am open for suggestions on providing better error messages.

Reply:
I think that is probably good enough for now. We can see if problems come up after merging.

Reply:
One idea (if this turns out to be a problem) could be to do a regexp comparison instead, ignoring the anonymous type parts of the names generated by the C# compiler.